app/frontend/wallet/shelley/transaction/utils.ts

```diff
@@ -1,53 +1,35 @@
 import BigNumber from 'bignumber.js'
-import distinct from '../../../helpers/distinct'
 import {Address, CertificateType, Lovelace, TokenBundle} from '../../../types'
-import {aggregateTokenBundles} from '../../helpers/tokenFormater'
+import {encodeCbor} from '../../helpers/cbor'
 import {TxCertificate, TxPlanAuxiliaryData, TxInput, TxOutput, TxWithdrawal} from '../../types'
-import {MIN_UTXO_VALUE} from './constants'
+import {cborizeSingleTxOutput} from '../shelley-transaction'
 import {estimateTxSize} from './estimateTxSize'
 
-export const computeMinUTxOLovelaceAmount = (tokenBundle: TokenBundle): Lovelace => {
-  // based on https://github.com/input-output-hk/cardano-ledger-specs/blob/master/doc/explanations/min-utxo.rst
-  const quot = (x: number, y: number) => Math.floor(x / y)
-  const roundupBytesToWords = (x: number) => quot(x + 7, 8)
-  // TODO: this to network config or constants
-  const minUTxOValue = new BigNumber(MIN_UTXO_VALUE)
-  // NOTE: should be 2, but a bug in Haskell set this to 0
-  const coinSize = 0
-  const txOutLenNoVal = 14
-  const txInLen = 7
-  const utxoEntrySizeWithoutVal = 6 + txOutLenNoVal + txInLen // 27
+export const computeMinUTxOLovelaceAmount = (
+  address: Address,
+  tokenBundle: TokenBundle
+): Lovelace => {
+  // Reference:
+  // - https://cips.cardano.org/cips/cip55/#thenewminimumlovelacecalculation
+  // - https://hydra.iohk.io/build/15339994/download/1/babbage-changes.pdf page 9
+  const overhead = 160
 
-  // NOTE: should be 29 but a bug in Haskell set this to 27
-  const adaOnlyUtxoSize = utxoEntrySizeWithoutVal + coinSize
+  // taken from https://cexplorer.io/params
+  const coinsPerUTxOByte = 4310
 
-  // this ensures there are only distinct (policyId, assetName) pairs
-  const aggregatedTokenBundle = aggregateTokenBundles([tokenBundle])
+  // 80kb is current max block size
+  const coinsUpperBound = new BigNumber(80 * 1024 * coinsPerUTxOByte) as Lovelace
 
-  const distinctAssets = aggregatedTokenBundle.map(({assetName}) => assetName)
-
-  const numAssets = distinctAssets.length
-  // number of unique policyIds
-  const numPIDs = distinct(aggregatedTokenBundle.map(({policyId}) => policyId)).length
-
-  const sumAssetNameLengths = distinctAssets.reduce(
-    (acc, assetName) => acc + Math.max(Buffer.from(assetName, 'hex').byteLength, 1),
-    0
-  )
-
-  const policyIdSize = 28 // pidSize in specs
+  const txOutput: TxOutput = {
+    isChange: false,
+    address,
+    coins: coinsUpperBound,
+    tokenBundle,
+  }
 
-  const size =
-    6 + roundupBytesToWords(numAssets * 12 + sumAssetNameLengths + numPIDs * policyIdSize)
+  const serSizeTxOut: number = encodeCbor(cborizeSingleTxOutput(txOutput)).length + 1
```

Comment:
```suggestion
  const serializedTxOutputSize: number = encodeCbor(cborizeSingleTxOutput(txOutput)).length + 1
```

(nit: is the explicit type needed?)

Reply:
encodeCbor is actually not properly typed, returning `any` - a proper fix would be to add an explicit type to that function and then this casting would no longer be needed it, I'll do it